src/modeler/AutoRest.Swagger/Validation/OperationParametersValidation.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class OperationParametersValidation : TypedRule<SwaggerParameter>
+    {
+        private const string SubscriptionId = "subscriptionId";
+        private const string ApiVersion = "api-version";
+        /// <summary>
+        /// This rule passes if the parameters are not subscriptionId or api-version
+        /// </summary>
+        /// <param name="paths"></param>
+        /// <returns></returns>
+        public override bool IsValid(SwaggerParameter Parameter)
+        {
```

Comment:
rather than performing the separate checks on the `Parameter` and `Parameter.Name`, you can use the null-propagation operator to fold it into a single check and let the compiler to the heavy lifting. 

``` csharp 
public override bool IsValid(SwaggerParameter Parameter) => 
    SubscriptionId  != Parameter?.Name && 
    ApiVersion != Parameter?.Name;
```